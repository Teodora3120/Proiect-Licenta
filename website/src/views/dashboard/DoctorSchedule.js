import {
  Container,
  Row,
  Col,
  Card,
  CardHeader,
  Label,
  CardFooter,
  CardBody,
  Input,
  Button,
} from 'reactstrap'
// core components
/* eslint-disable react-hooks/exhaustive-deps */
import { useState, useEffect } from 'react'
import Header from 'components/Headers/Header'
import { useUserContext } from 'context/UserContext'
import ScheduleApi from 'api/schedules'

const DoctorSchedule = () => {
  const [days, setDays] = useState([
    { id: '', dayOfWeek: 'Monday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Tuesday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Wednesday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Thursday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Friday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Saturday', startTime: '', endTime: '' },
    { id: '', dayOfWeek: 'Sunday', startTime: '', endTime: '' },
  ])
  const [error, setError] = useState(null)
  const [success, setSuccess] = useState(false)
  const { user } = useUserContext()
  let timeout

  useEffect(() => {
    const fetchData = async () => {
      await getSchedule()
    }
    fetchData()
  }, [])

  const getSchedule = async () => {
    try {
      const response = await ScheduleApi.GetSchedule(user.id)
      setDays(response.data)
    } catch (err) {
      setError('Server error.')
    }
  }
  const updateState = (index, target, value) => {
    setDays(
      days.map((day) => {
        if (days.indexOf(day) === index) {
          if (target === 'start') {
            return { ...day, startTime: value }
          } else if (target === 'end') {
            return { ...day, endTime: value }
          }
        }
        return day
      }),
    )
  }

  const sendSchedule = async () => {
    try {
      const data = days.map((item) => {
        return {
          id: item.id,
          startTime: item.startTime,
          endTime: item.endTime,
        }
      })
      await ScheduleApi.SendSchedule(data)
      await getSchedule()
      setSuccess(true)
      timeout = setTimeout(() => {
        setSuccess(false)
      }, 2000)
    } catch (error) {
      setError('Server error.')
    }
  }

  useEffect(() => {
    setError(null)
  }, [days])

  useEffect(() => {
    return () => {
      clearTimeout(timeout)
    }
  }, [])

  return (
    <>
      <Header />
      {/* Page content */}
      <Container className="mt--7" fluid>
        {/* Table */}
        <Row>
          <Col>
            <Card className="shadow">
              <CardHeader>
                <Row className="align-items-center">
                  <Col lg="8" md="8" sm="12" xs="12" className="text-left">
                    <h3 className="mb-0">
                      Set your schedule for the current week
                    </h3>
                  </Col>
                  <Col
                    lg="4"
                    md="4"
                    sm="12"
                    xs="12"
                    className="text-right d-flex align-items-center justify-content-end"
                  >
                    <h3
                      style={{ transition: '.2s all' }}
                      className={`text-right text-success mb-0 mr-3 font-weight-600 ${success ? 'd-initial' : 'd-none'
                        }`}
                    >
                      Success!
                    </h3>
                    {error ? (
                      <h4 className="text-right text-danger mb-0 mr-3 font-weight-400">
                        {error}
                      </h4>
                    ) : null}
                    <Button className='ws-0' onClick={sendSchedule} color="primary">
                      Save Schedule
                    </Button>
                  </Col>
                </Row>
              </CardHeader>
              <CardBody>
                <Row>
                  {days.map((day, index) => {
                    return (
                      <Col
                        lg="4"
                        md="6"
                        sm="6"
                        xs="12"
                        className="mb-4"
                        key={index}
                      >
                        <Card className="bg-secondary shadow">
                          <CardBody>
                            <Row>
                              <Col className="text-center">
                                <h2>{day.dayOfWeek}</h2>
                                <hr />
                                <Row>
                                  <Col
                                    xl="6"
                                    lg="12"
                                    md="12"
                                    sm="6"
                                    xs="6"
                                    className="text-left"
                                  >
                                    <Label>Start time</Label>
                                    <Input
                                      defaultValue={day.startTime}
                                      onChange={(e) =>
                                        updateState(
                                          index,
                                          'start',
                                          e.target.value,
                                        )
                                      }
                                      type="time"
                                    />
                                  </Col>
                                  <Col
                                    xl="6"
                                    lg="12"
                                    md="12"
                                    sm="6"
                                    xs="6"
                                    className="text-left"
                                  >
                                    <Label>End time</Label>
                                    <Input
                                      defaultValue={day.endTime}
                                      onChange={(e) =>
                                        updateState(
                                          index,
                                          'end',
                                          e.target.value,
                                        )
                                      }
                                      type="time"
                                    />
                                  </Col>
                                </Row>
                              </Col>
                            </Row>
                          </CardBody>
                        </Card>
                      </Col>
                    )
                  })}
                </Row>
              </CardBody>
              <CardFooter className="border-0 pt-0">
                <Row>
                  <Col className="text-center"></Col>
                </Row>
              </CardFooter>
            </Card>
          </Col>
        </Row>
      </Container>
    </>
  )
}

export default DoctorSchedule
