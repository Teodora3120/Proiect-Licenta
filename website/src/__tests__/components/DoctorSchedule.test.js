import React from 'react'
import { shallow } from 'enzyme'
import DoctorSchedule from '../../views/dashboard/DoctorSchedule'
it('should test DoctorSchedule component', () => {
  const wrapper = shallow(< DoctorSchedule />)
  expect(wrapper).toMatchSnapshot()
})