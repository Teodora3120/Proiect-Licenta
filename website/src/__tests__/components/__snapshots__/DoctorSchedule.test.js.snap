// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test DoctorSchedule component 1`] = `
<Fragment>
  <Header />
  <Container
    className="mt--7"
    fluid={true}
    tag="div"
  >
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Card
          className="shadow"
          tag="div"
        >
          <CardHeader
            tag="div"
          >
            <Row
              className="align-items-center"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Col
                className="text-left"
                lg="8"
                md="8"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <h3
                  className="mb-0"
                >
                  Set your schedule for the current week
                </h3>
              </Col>
              <Col
                className="text-right d-flex align-items-center justify-content-end"
                lg="4"
                md="4"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <h3
                  className="text-right text-success mb-0 mr-3 font-weight-600 d-none"
                  style={
                    Object {
                      "transition": ".2s all",
                    }
                  }
                >
                  Success!
                </h3>
                <Button
                  className="ws-0"
                  color="primary"
                  onClick={[Function]}
                  tag="button"
                >
                  Save Schedule
                </Button>
              </Col>
            </Row>
          </CardHeader>
          <CardBody
            tag="div"
          >
            <Row
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Col
                className="mb-4"
                key="0"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Monday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="1"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Tuesday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="2"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Wednesday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="3"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Thursday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="4"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Friday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="5"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Saturday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
              <Col
                className="mb-4"
                key="6"
                lg="4"
                md="6"
                sm="6"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="12"
              >
                <Card
                  className="bg-secondary shadow"
                  tag="div"
                >
                  <CardBody
                    tag="div"
                  >
                    <Row
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <Col
                        className="text-center"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <h2>
                          Sunday
                        </h2>
                        <hr />
                        <Row
                          tag="div"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              Start time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                          <Col
                            className="text-left"
                            lg="12"
                            md="12"
                            sm="6"
                            tag="div"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                            xl="6"
                            xs="6"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              End time
                            </Label>
                            <Input
                              defaultValue=""
                              onChange={[Function]}
                              type="time"
                            />
                          </Col>
                        </Row>
                      </Col>
                    </Row>
                  </CardBody>
                </Card>
              </Col>
            </Row>
          </CardBody>
          <CardFooter
            className="border-0 pt-0"
            tag="div"
          >
            <Row
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Col
                className="text-center"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              />
            </Row>
          </CardFooter>
        </Card>
      </Col>
    </Row>
  </Container>
</Fragment>
`;
